lib/ash_phoenix/form/form.ex

```diff
@@ -1347,12 +1347,8 @@ defmodule AshPhoenix.Form do
                  Enum.all?(pkey_fields, &Map.has_key?(params, to_string(&1))) do
               Enum.all?(pkey_fields, fn field ->
                 case Map.fetch(params, to_string(field)) do
-                  {:ok, v} ->
-                    v ==
-                      AshPhoenix.Form.value(nested_form, field)
-
-                  _ ->
-                    false
+                  {:ok, v} -> to_string(v) == to_string(AshPhoenix.Form.value(nested_form, field))
```

Comment:
There should be a case where we check if they are equal first. Then, we only convert either one to a string if the other one is a string and then compare.